boring/src/ssl/mod.rs

```diff
@@ -1960,7 +1960,7 @@ impl SslContextBuilder {
     /// threads.
     #[cfg(not(feature = "fips"))]
     #[corresponds(SSL_CTX_set1_ech_keys)]
-    pub fn set_ech_keys(&mut self, keys: SslEchKeys) -> Result<(), ErrorStack> {
+    pub fn set_ech_keys(&self, keys: SslEchKeys) -> Result<(), ErrorStack> {
         unsafe { cvt(ffi::SSL_CTX_set1_ech_keys(self.as_ptr(), keys.as_ptr())).map(|_| ()) }
```

Comment:
Why doesn't this take a mut ref?

Reply:
I suppose it's fine as we're just passing an opaque value into OpenSSL which handles the allocation, right?

Reply:
This API is meant to be called after the SSL_CTX has presumably been distributed to multiple threads, so it's thread safe under the hood. 

We take ownership of `keys` and drop it at the end of the function to shed our refcount and let boringssl take care of the rest